server/src/main/java/org/elasticsearch/index/mapper/MetadataFieldMapper.java

```diff
@@ -109,6 +115,11 @@ public Builder parse(String name, Map<String, Object> node, MappingParserContext
         public MetadataFieldMapper getDefault(MappingParserContext parserContext) {
             return defaultMapperParser.apply(parserContext);
         }
+
+        @Override
+        public boolean supportsVersion(Version indexCreatedVersion) {
+            return true;
```

Comment:
I wonder if overriding these two supportsVersion is needed. Maybe it does because metadata fields are always supported and they don't take a version in like FieldMapper does?

Reply:
This was never called, it mainly resulted from the fact that MetadataFieldMapper.TypeParser was extending Mapper.TypeParser (with no good reason). I separated the two in [b25369e](https://github.com/elastic/elasticsearch/pull/85059/commits/b25369e1060ba5709fee15cbec9d776e45a34ac1), which means we no longer have to have this method here.